### PR TITLE
Add an example of option pricing using Black-Scholes equation

### PR DESCRIPTION
**Pull request by rezoo**
This PR adds an example of option pricing using Black-Scholes equation and CuPy.
I created this example in order to illustrate how to write our own kernels with CuPy and its performance.

In my environment (CPU: Intel(R) Xeon(R) CPU E5-2623 v3 @ 3.00GHz, GPU: TITAN X(Pascal)), These performances are the following:

```
 CPU (NumPy, Naive implementation):     19.997561 sec
 GPU (CuPy, Naive implementation):      0.319212 sec
 GPU (CuPy, Elementwise kernel):        0.012139 sec
```

### CODE REVIEW

File: examples/finance/black_scholes.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+    print('start computation')
+    risk_free = 0.02
+    volatility = 0.3
+    with timer(' CPU (NumPy, Naive implementation)'):
+        call_cpu, put_cpu = black_scholes(
+            numpy, stock_price_cpu, option_strike_cpu, option_years_cpu,
+            risk_free, volatility)
+
+    with timer(' GPU (CuPy, Naive implementation)'):
+        call_cpu, put_cpu = black_scholes(
```
Comment (gwtnb) [inline examples/finance/black_scholes.py]: `call_gpu, put_gpu` ?

File: examples/finance/black_scholes.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+import argparse
+import contextlib
+import time
+
+import cupy
+import numpy
+
+# This sample computes call and put prices for Europian options with
+# Black-Scholes equation. It was based on a sample of the financial package
+# in CUDA toolkit. For details, please see a corresponding whitepaper.
```
Comment (gwtnb) [inline examples/finance/black_scholes.py]: the corresponding whitepaper

**Comment by rezoo** [comment]
Thanks. I fixed.

File: examples/finance/black_scholes.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+#
+# The following code shows that CuPy enables us to write algorithms for GPUs
+# without significantly modifying the existing NumPy's code.
+# It also briefly describes how to create your own kernel with CuPy.
+# If you want to speed up the existing code, please define the kernel
+# with cupy.ElementwiseKernel.
+
+
+# Naive implementation of the pricing of options with NumPy.
+def black_scholes(xp, s, x, t, r, v):
```
Comment (niboshi) [inline examples/finance/black_scholes.py]: This function is also run with cupy, so it's better to mention it as well.

Comment (rezoo) [reply examples/finance/black_scholes.py]: Fixed.

**Comment by gwtnb** [comment]
LGTM

**Pull request merged**